src/addresses.rs

```diff
@@ -136,18 +208,37 @@ pub fn find_address(
                 geofinder.get(&poi.coord),
             ))
         }
-        _ => rubber
-            .get_address(&poi.coord)
-            .map_err(|e| {
-                warn!("get_address returned ES error for {}: {}", poi.id, e);
-                e
-            })
-            .ok()
-            .and_then(|addrs| addrs.into_iter().next())
-            .map(|addr| {
-                addr.address()
-                    .expect("get_address returned a non-address object")
-            }),
+        _ => {
+            if !addr_updated {
+                // Fetch the address already attached to the POI to avoid computing an unnecessary
+                // reverse.
+                match get_current_addr(rubber, poi_index, &poi.id) {
+                    CurPoiAddress::None => {}
+                    CurPoiAddress::NotFound => return None,
```

Comment:
Is that not supposed to be the opposite ? `NotFound` corresponds to POIs that don't exist in the previous index and for which a reverse is always necessary.

Also, the comparison of coords should also be checked when the address is `None`. In this case too the position may have changed, and a reverse could be necessary.

Reply:
That was right, I just fixed this.

Maybe I should add some tests about this, as it's already pretty hard to do manually.